lib/rules/no-static-imports-of-user-specific-scoped-modules-in-ssrable-components.js

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2024, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+'use strict';
+
+const { docUrl } = require('../util/doc-url');
+
+const userScopedModulesSet = new Set([
+    '@salesforce/user',
+    '@salesforce/userPermission',
+    '@salesforce/customPermission',
+    '@salesforce/featureFlag',
+]);
+
+function isUserScopedModule(importSource) {
+    return [...userScopedModulesSet].some((module) => importSource.startsWith(module));
+}
+
+module.exports = {
+    meta: {
+        type: 'problem',
+        docs: {
+            description:
+                'Disallow static imports of user-specific scoped modules in SSR-able components',
+            category: 'LWC',
+            url: docUrl('no-static-imports-of-user-specific-scoped-modules-in-ssrable-components'),
+            recommended: true,
+        },
+        messages: {
+            noStaticUserScopedImports:
+                'Static import of @salesforce user-specific scoped modules is not allowed in SSR-able components. The recommended declarative solution is to use a data provider to fetch this information. The dynamic import approach is an anti-pattern and should be avoided.',
```

Comment:
Should we remove the dynamic import bit and have a followup PR which adds a warning for all dynamic imports during SSR?
```suggestion
                'Static import of @salesforce user-specific scoped modules is not allowed in SSR-able components. The recommended declarative solution is to use a data provider to fetch this information.',
```

Reply:
I'm fine with that as long as everyone is aligned with this suggestion.